src/main/scala/shine/DPIA/InferAccessAnnotation.scala

```
@@ -133,6 +132,9 @@ private class InferAccessAnnotation {
         inferDepLambda(depL, ctx, isKernelParamFun)
       case depA: r.DepApp[_] =>
         inferDepApp(depA, ctx, addsKernelParam(e, isKernelParamFun))
+      case TypeAnnotation(e, _) => inferPhraseTypes(e, ctx, isKernelParamFun)
+      case TypeAssertion(e, _) => inferPhraseTypes(e, ctx, isKernelParamFun)
+      case Opaque(e, _) => inferPhraseTypes(e, ctx, isKernelParamFun)
```

these should throw an exception?

---

Fixed